specs/plugin-extensibility.md

```diff
@@ -155,53 +171,119 @@ This interface targets plugins that integrate with providers of basic cryptograp
 
 #### Signing workflow using plugin
 
-1. Given a user request to sign `oci-artifact`, with `keyName` (the friendly key name)
-1. Pull the image manifest using `oci-artifact` url, and construct a descriptor
-1. Append any user provided metadata and Notary metadata as descriptor annotations.
-1. Determine if the registered key uses a plugin
-1. Execute the plugin with `get-plugin-metadata` command
+1. Given a user request to sign `oci-artifact`, with signing key `keyName` (the friendly key name)
+2. Pull the image manifest using `oci-artifact` url, and construct a descriptor
+3. Append any user provided metadata and Notary metadata as descriptor annotations.
+4. Determine if the registered key uses a plugin
+5. Execute the plugin with `get-plugin-metadata` command
     1. If plugin supports capability `SIGNATURE_GENERATOR`
-        1. Generate the payload to be signed. For JWS this includes [JWS](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#supported-signature-envelopes) payload with `subject` claim as descriptor, additional JWS claims, and protected headers.
-        2. Execute the plugin with `generate-signature` command, set `request.keyDefinition` to key definition corresponding to `keyName`, `request.payload` to the generated payload.
-        3. Validate the generated signature, return an error if any of the checks fails.
+        1. Execute the plugin with `describe-key` command, set `request.keyId` and the optional `request.pluginConfig` to corresponding values associated with signing key `keyName` in `config.json`.
+        2. Generate the payload to be signed for [JWS](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#supported-signature-envelopes) envelope format.
+           1. Create the JWS protected headers collection and set `alg` to value corresponding to `describe-key.response.keySpec` as per [signature algorithm selection](https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#algorithm-selection).
+           2. Create the `JWSPayload` with appropriate private (`subject`) and public (`iat,exp`) claims.
```

Comment:
How does notation set iat, exp values? Wouldn't these be dependent on the plugin itself?

Reply:
`iat` (signature creation time) and `exp` (optional expiry time based on user input) will be set by notation, these are not influenced by the key itself when using the `signature-generator` interface.